yt_dlp/extractor/panopto.py

```diff
@@ -0,0 +1,444 @@
+                'title': timestamp.get('Caption')
+            })
+        return chapters
+
+    def _extract_streams_formats_and_subtitles(self, video_id, streams, **fmt_kwargs):
+        formats = []
+        subtitles = {}
+        for stream in streams or []:
+            stream_formats = []
+            http_stream_url = stream.get('StreamHttpUrl')
+            stream_url = stream.get('StreamUrl')
+
+            if http_stream_url:
+                stream_formats.append({'url': http_stream_url})
+
+            if stream_url:
+                media_type = stream.get('ViewerMediaFileTypeName')
+                if media_type in ('hls', ):
+                    m3u8_formats, stream_subtitles = self._extract_m3u8_formats_and_subtitles(stream_url, video_id)
+                    stream_formats.extend(m3u8_formats)
+                    subtitles = self._merge_subtitles(subtitles, stream_subtitles)
+                else:
+                    stream_formats.append({
+                        'url': stream_url
+                    })
+            for fmt in stream_formats:
+                fmt.update({
+                    'format_note': stream.get('Tag'),
+                    **fmt_kwargs
+                })
+            formats.extend(stream_formats)
+
+        return formats, subtitles
+
+    def _real_extract(self, url):
+        base_url, video_id = self._match_valid_url(url).group('base_url', 'id')
+        delivery_info = self._call_api(
+            base_url, '/Pages/Viewer/DeliveryInfo.aspx', video_id,
+            query={
+                'deliveryId': video_id,
+                'invocationId': '',
+                'isLiveNotes': 'false',
+                'refreshAuthCookie': 'true',
+                'isActiveBroadcast': 'false',
+                'isEditing': 'false',
+                'isKollectiveAgentInstalled': 'false',
+                'isEmbed': 'false',
+                'responseType': 'json',
+            }
+        )
+
+        delivery = delivery_info['Delivery']
+        session_start_time = int_or_none(delivery.get('SessionStartTime'))
+
+        # Podcast stream is usually the combined streams. We will prefer that by default.
+        podcast_formats, podcast_subtitles = self._extract_streams_formats_and_subtitles(
+            video_id, delivery.get('PodcastStreams'), format_note='PODCAST')
+
+        streams_formats, streams_subtitles = self._extract_streams_formats_and_subtitles(
+            video_id, delivery.get('Streams'), preference=-10)
+
+        formats = podcast_formats + streams_formats
+        subtitles = self._merge_subtitles(podcast_subtitles, streams_subtitles)
+        self._sort_formats(formats)
+
+        self.mark_watched(base_url, video_id, delivery_info)
+
+        return {
+            'id': video_id,
+            'title': delivery.get('SessionName'),
+            'cast': list(filter(None, traverse_obj(delivery, ('Contributors', ..., 'DisplayName'), default=[]))),
```

Comment:
```suggestion
            'cast': traverse_obj(delivery, ('Contributors', ..., 'DisplayName')),
```
just this should be enough, no?

Reply:
DisplayName may exist but it's value may be empty

E.g. first test case:
![image](https://user-images.githubusercontent.com/27879174/157185222-2afd3a5a-8440-4867-9193-3d24a21e627e.png)

Reply:
`expected_type=strip_or_none`

Reply:
won't work, since strip_or_none will return an empty string